xwiki-platform-core/xwiki-platform-livetable/xwiki-platform-livetable-ui/src/main/resources/XWiki/LiveTableDBListFilterService.xml

```diff
@@ -0,0 +1,101 @@
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.3" reference="XWiki.LiveTableDBListFilterService" locale="">
+  <web>XWiki</web>
+  <name>LiveTableDBListFilterService</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <creationDate>1496336171000</creationDate>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <date>1497000123000</date>
+  <contentUpdateDate>1496999359000</contentUpdateDate>
+  <version>1.1</version>
+  <title/>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>false</hidden>
+  <content>{{velocity wiki="false"}}
+#set($reqClass = "$!request.get('class')")
+#set($reqProp = "$!request.get('prop')")
+#set($reqInput = "$!request.get('input')")
+#set($reqLimit = "$!request.get('limit')")
+#if ($reqLimit != '')
+  #set ($limit = $numbertool.integer($reqLimit))
+#else
+  ## The limit is hardcoded here, as we are only dealing with suggestions
+  #set ($limit = 25)
+#end
+#if ($reqClass != '' &amp;&amp; $reqProp != '' &amp;&amp; $reqInput != '')
+  #set ($discard = $response.setContentType('application/json'))
+  #set ($classRef = $xwiki.getClass($reqClass))
+  #set ($field = $classRef.get($reqProp))
+  #if ($field.classType == 'DBList')
+    #set ($sql = $!field.getProperty('sql').value)
+    #set ($results = [])
+    #if ($sql == '')
+      ## Use SOLR
+      #set ($dbId = "$!field.getProperty('idField').value")
+      #set ($dbValue = "$!field.getProperty('valueField').value")
+      #set ($dbClass = $services.model.serialize("$!field.getProperty('classname').value", 'solr'))
+      #set ($solr = "property.${dbClass}.${dbValue}:$reqInput")
+      #set ($query = $services.query.createQuery($solr, 'solr'))
+      #set ($discard = $query.bindValue('fq', "type:DOCUMENT class:$dbClass"))
+      #set ($discard = $query.setLimit($limit))
+      #foreach ($obj in $query.execute()[0].results)
+        #set ($res = {
+          'id': $obj.getFieldValue("property.${dbClass}.${dbId}_string"),
+          'name': $obj.getFieldValue("property.${dbClass}.${dbValue}_string")
+        })
+        #set($discard = $results.add($res))
+      #end
+    #else
+      ## Use HQL as there is a query present
+      #set ($sql = $sql + " AND UPPER(doc.title) LIKE UPPER(:input)")
```

Comment:
You assume that the display value is the page title, but that's not always true. What you know is that the rows selected by the query have two columns, the first is the id and the second is the display value. So you need to filter on the second column, or even both. Moreover, the HQL query might end with an ``order by`` clause or it may not even have the ``where`` clause, so you cannot simply append the ``AND`` constraint at the end.